homeassistant/components/aqara/binary_sensor.py

```diff
@@ -0,0 +1,221 @@
+    hass_data: HomeAssistantAqaraData = hass.data[DOMAIN][entry.entry_id]
+
+    def async_add_battery_low_entity(device_id, res_id) -> None:
+        """Add battery low entity by event. call from other component."""
+        aqara_point = hass_data.device_manager.device_map[device_id].point_map.get(
+            hass_data.device_manager.make_point_id(device_id, res_id)
+        )
+        if aqara_point is not None:
+            entities: list[AqaraBinarySensorEntity] = [
+                AqaraBinarySensorEntity(
+                    aqara_point, hass_data.device_manager, battery_staus_desc
+                )
+            ]
+            async_add_entities(entities)
+
+    @callback
+    def async_discover_device(device_ids: list[str]) -> None:
+        """Discover and add a discovered Aqara binary sensor."""
+        entities: list[AqaraBinarySensorEntity] = []
+
+        def append_entity(aqara_point, description):
+            entities.append(
+                AqaraBinarySensorEntity(
+                    aqara_point, hass_data.device_manager, description
+                )
+            )
+
+        find_aqara_device_points_and_register(
+            hass, entry.entry_id, hass_data, device_ids, BINARY_SENSORS, append_entity
+        )
+        async_add_entities(entities)
+
+    async_discover_device([*hass_data.device_manager.device_map])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, AQARA_DISCOVERY_NEW, async_discover_device)
+    )
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, AQARA_BATTERY_LOW_ENTITY_NEW, async_add_battery_low_entity
+        )
+    )
+
+
+class AqaraBinarySensorEntity(AqaraEntity, BinarySensorEntity):
+    """Aqara Binary Sensor Entity."""
+
+    entity_description: AqaraBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        point: AqaraPoint,
+        device_manager: AqaraDeviceManager,
+        description: AqaraBinarySensorEntityDescription,
+    ) -> None:
+        """Init Aqara binary sensor."""
+        super().__init__(point, device_manager)
+        self.entity_description = description
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if sensor is on."""
+        return (
+            self.point.get_value() == self.entity_description.on_value
+            or self.point.get_value() in self.entity_description.on_value
+        )
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon to use in the frontend, if any."""
+        if hasattr(self, "_attr_icon"):
+            return self._attr_icon
+        if hasattr(self, "entity_description"):
+            if self.is_on is True and self.entity_description.on_icon != "":
+                return self.entity_description.on_icon
+            if self.is_on is False and self.entity_description.off_icon != "":
+                return self.entity_description.off_icon
+
+            return self.entity_description.icon
+        return None
```

Comment:
This looks like the default implementation? And thus can be removed?

Reply:
removed